### ISSUE TITLE
support protobuf versions older than 3.4

### ISSUE THREAD

**Issue by nolange** [grpc/grpc#14838]
Hello,

I tried to compile grpc with the protobuf version 3.0 from debian testing. Seems like 2 spots require methods introduced with protobuf 3.4.

The patch below fixes the build

```diff
--- grpc-1.10.orig/src/compiler/php_generator.cc
+++ grpc-1.10/src/compiler/php_generator.cc
@@ -44,9 +44,13 @@ grpc::string ConvertToPhpNamespace(const
 }
 
 grpc::string PackageName(const FileDescriptor* file) {
+#if GOOGLE_PROTOBUF_VERSION >= 3004000
   if (file->options().has_php_namespace()) {
     return file->options().php_namespace();
   } else {
+#else
+{
+#endif
     return ConvertToPhpNamespace(file->package());
   }
 }
--- grpc-1.10.orig/src/compiler/php_generator_helpers.h
+++ grpc-1.10/src/compiler/php_generator_helpers.h
@@ -48,9 +48,13 @@ inline grpc::string GetPHPServiceFilenam
     const grpc::protobuf::ServiceDescriptor* service,
     const grpc::string& class_suffix) {
   std::ostringstream oss;
+#if GOOGLE_PROTOBUF_VERSION >= 3004000
   if (file->options().has_php_namespace()) {
     oss << ReplaceAll(file->options().php_namespace(), "\\", "/");
   } else {
+#else
+  {
+#endif
     std::vector<grpc::string> tokens =
         grpc_generator::tokenize(file->package(), ".");
     for (unsigned int i = 0; i < tokens.size(); i++) {
```

**Comment by nolange** [comment]
You claim to require only protobuf 3.0 in your released version 1.10, so i consider it a bug if grpc does not work.
I can and want to use more recent versions, but i would also like to just be able to use the packaged version from my distribution, and this does not seem to be updated any time soon.

**Comment by srini100** [comment]
Right, 3.0 is the min. version except for PHP plugin. Since top level make builds multiple langs and users are running into build issues, we decided to bump the protobuf version check to 3.5 in the upcoming 1.11 release. The document will reflect that. Closing this.
